docs/repo-docs/crafting-your-repository/constructing-ci.mdx

```diff
@@ -54,7 +56,22 @@ For more on how to deploy from a Turborepo with Docker, visit [the dedicated Doc
 
 ## Skipping tasks and other unnecessary work
 
-Using Turborepo's built-in caching is a great way to speed up your CI. We encourage you to start by relying on Turborepo's caching and parallelization when you're starting out with Turborepo.
+### Running only affected tasks
+
+You can use the `--affected` flag to only run tasks that have changes.
+
+```bash title="Terminal
+turbo run build --affected
+```
+
+You'll want to use this flag in situations like:
+
+- You're running many tasks across packages in your monorepo, and only want to run those tasks in packages with code changes.
+- You’re _not_ using a Remote Cache, but still want to do as little work as possible in CI.
+- You _are_ using a Remote Cache, and you’re in a large repository. By minimizing the amount of tasks that will be restored from cache, there will be less data to send across the network, resulting in faster cache restoration.
+- You’re already using [advanced filtering techniques](/repo/docs/reference/run#advanced-filtering-examples) or [`turbo-ignore`](/repo/docs/reference/turbo-ignore) to create the same or similar behavior as `--affected`. You likely have the opportunity to simplify your scripting using this new flag.
```

Comment:
One big benefit is when you're using a shallow clone. With the normal filter syntax, if the base doesn't exist in the shallow clone, turbo crashes. With affected, we gracefully fall back to running all the tasks